src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpWindowsProxy.cs

```diff
@@ -186,84 +216,127 @@ public void Dispose()
         /// or whitespace separated list, with each entry in the following format:
         /// ([&lt;scheme&gt;=][&lt;scheme&gt;"://"]&lt;server&gt;[":"&lt;port&gt;])
         /// </summary>
-        private static void ParseProxyConfig(string value, out Uri insecureProxy, out Uri secureProxy)
+        private static void ParseProxyConfig(ReadOnlySpan<char> proxyString, out MultiProxy secureProxy, out MultiProxy insecureProxy)
         {
-            insecureProxy = null;
-            secureProxy = null;
-            if (string.IsNullOrEmpty(value))
-            {
-                return;
-            }
+            Uri[] secureUris = new Uri[4], insecureUris = new Uri[4];
```

Comment:
What's the rationale behind starting these both at an array size of 4?  How common will it be to have more than one Uri in each?

Regardless of what size we choose, a comment explaining it would be helpful.

Reply:
I assume not common, but have no data. We can start it at 1 to optimize for this case. Growing by 1 might actually be appropriate if we assume the next common case will be 2.

Reply:
Updated to not allocate arrays, except for init-once system-wide proxy, which grows by 1.